pydicom/jsonrep.py

```diff
@@ -4,194 +4,278 @@
 import base64
 from inspect import signature
 import inspect
-from typing import Callable, Optional, Union, Any, cast
+from typing import (
+    Callable, Optional, Union, Any, cast, Type, TypeVar, Dict, TYPE_CHECKING,
+    List
+)
 import warnings
 
-from pydicom.tag import BaseTag
+if TYPE_CHECKING:  # pragma: no cover
+    from pydicom.dataset import Dataset
 
 # Order of keys is significant!
 JSON_VALUE_KEYS = ('Value', 'BulkDataURI', 'InlineBinary',)
 
-BINARY_VR_VALUES = ['OW', 'OB', 'OD', 'OF', 'OL', 'UN',
-                    'OB or OW', 'US or OW', 'US or SS or OW']
-VRs_TO_BE_FLOATS = ['DS', 'FL', 'FD', ]
-VRs_TO_BE_INTS = ['IS', 'SL', 'SS', 'UL', 'US', 'US or SS']
+BINARY_VR_VALUES = [
+    'OB', 'OD', 'OF', 'OL', 'OV', 'OW', 'UN',
+    'OB or OW', 'US or OW', 'US or SS or OW'
+]
+VRs_TO_BE_FLOATS = ['DS', 'FD', 'FL']
+VRs_TO_BE_INTS = ['IS', 'SL', 'SS', 'SV', 'UL', 'US', 'UV', 'US or SS']
 
 
-def convert_to_python_number(value, vr):
-    """Makes sure that values are either ints or floats
+def convert_to_python_number(value: Any, vr: str) -> Any:
+    """When possible convert numeric-like values to either ints or floats
     based on their value representation.
 
     .. versionadded:: 1.4
 
     Parameters
     ----------
-    value: Union[Union[str, int, float], List[Union[str, int, float]]]
-        value of data element
-    vr: str
-        value representation of data element
+    value : Any
+        Value of the data element.
+    vr : str
+        Value representation of the data element.
 
     Returns
     -------
-    Union[Union[str, int, float], List[Union[str, int, float]]]
+    Any
+
+        * If `value` is empty then returns the `value` unchanged.
+        * If `vr` is an integer-like VR type then returns ``int`` or
+          ``List[int]``
+        * If `vr` is a float-like VR type then returns ``float`` or
+          ``List[float]``
+        * Otherwise returns `value` unchanged
 
     """
-    if value is None:
-        return None
-    number_type = None
+    from pydicom.dataelem import empty_value_for_VR
+
+    if value is None or "":
+        return value
+
+    number_type: Optional[Union[Type[int], Type[float]]] = None
     if vr in VRs_TO_BE_INTS:
         number_type = int
     if vr in VRs_TO_BE_FLOATS:
         number_type = float
-    if number_type is not None:
-        if isinstance(value, (list, tuple,)):
-            value = [number_type(e) for e in value]
-        else:
-            value = number_type(value)
-    return value
+
+    if number_type is None:
+        return value
+
+    if isinstance(value, (list, tuple)):
+        return [
+            number_type(v) if v is not None
+            else empty_value_for_VR(vr)
+            for v in value
+        ]
```

Comment:
[Standard says there may be a null value in the list](http://dicom.nema.org/medical/dicom/current/output/chtml/part18/sect_F.2.5.html)